Changes neutral mode to brake default

JackTWing opened the pull request.

JackTWing committed:
Merge pull request #3 from Team1100/master

Pulling changes JT 3-25-21

JackTWing committed:
Merge pull request #4 from Team1100/master

Pulling Changes 4-8-21   JT

JackTWing committed:
Merge pull request #5 from Team1100/master

Pulling changes 4-15-21 JT

JackTWing committed:
Merge pull request #6 from Team1100/master

Pulling Changes 5/6/21 JT

JackTWing committed:
Changes neutral mode to brake

This change configures the motors in the drive subsystem
to operate in braking mode.

JackTWing committed:
Merge pull request #7 from Team1100/master

Pulling changes 6-10-21 JT

JackTWing committed:
Merge pull request #8 from Team1100/master

Pulling Changes Jack Tommaney 9/9/21

JackTWing committed:
Merge pull request #9 from Team1100/master

Pulling Changes 9/23/21 JT

JackTWing committed:
adds ToggleBrakeMode command to the Drive subsystem

ToggleBrakeMode toggles the robot's driving brakes;
Left joystick button 3 turns it on, while Right joystick
button 3 turns it off.

JackTWing committed:
revises Drive.java logic operator for Brake Mode

Changed if(Brake = True) and (Brake = False) to
(Brake == True) and (Brake == False)

snowpuppy commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -166,6 +171,20 @@ public Encoder getLeftEncoder() {
+   
+    Brake = bm;
+
+    if (Brake == true) {backLeft.setNeutralMode(NeutralMode.Brake);
```
Do you mind fixing the indentation here?

snowpuppy commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -166,6 +171,20 @@ public Encoder getLeftEncoder() {
     return rightEncoder;
   }
+
+  public void brakeModeToggle(boolean bm) {
```
Do you think you could change the name to "brakeModeSet"? A toggle implies that no parameter is needed and that the state will flip back and forth every time the function is called.

snowpuppy commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -41,6 +42,8 @@
 
   private Encoder leftEncoder, rightEncoder;
 
+  public boolean Brake = true;               // SHOULD USUALLY BE FALSE!
```
Do you mind starting this variable with a lower case letter (i.e "brake" instead of "Brake")? Starting with an upper case letter is normally reserved for class names.

snowpuppy reviewed (commented):
I've left a few comments about style.

JackTWing committed:
Merge pull request #10 from Team1100/master

Pulling recent changes 11/4/21

JackTWing committed:
adds BrakeMode and implements it in Drive.java

added BrakeMode command, and uses BrakeModeSet in Drive.java, which can
be toggled by pressing a side button on the joystick.
